lib/system/sysio.nim

```diff
@@ -52,6 +52,8 @@ proc c_ferror(f: File): cint {.
   importc: "ferror", header: "<stdio.h>", tags: [].}
 proc c_setvbuf(f: File, buf: pointer, mode: cint, size: csize): cint {.
   importc: "setvbuf", header: "<stdio.h>", tags: [].}
+proc c_ftruncate(f: FileHandle, len: clong): cint {.
```

Comment:
I'm fine with reverting the sysio changes, so long as the platform specific ``AsyncFile`` stuff remains.